pkg/asset/agent/manifests/infraenv.go

```diff
@@ -36,49 +39,49 @@ func (*InfraEnv) Name() string {
 func (*InfraEnv) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&agent.OptionalInstallConfig{},
-		// &AgentPullSecret{},
```

Comment:
Not sure if I missed some previous discussion, but why the pull secret and nmstateconfig are missing here?

Reply:
This causes a dirty dependency issue and image generation fails with 
```
WARNING       Discarding the InfraEnv Config that was provided in the target directory
because its dependencies are dirty and it needs to be regenerated.
............
............
FATAL failed to fetch Agent Installer ISO: failed to fetch dependency of "Agent Installer ISO":
 failed to fetch dependency of "Agent Installer Ignition": 
failed to fetch dependency of "Agent Manifests": 
failed to generate asset "InfraEnv Config": missing configuration or manifest file
```
Ref https://github.com/openshift/installer/pull/6134#discussion_r925052704

Reply:
The installer effectively doesn't allow dependencies between stuff at the same level (e.g. all stuff created by `agent create cluster-manifests`). You can depend on stuff generated by the previous command, but for stuff generated by the same command it will overwrite the user's changes if there is a dependency between them.
If it looked at file modification times or something (like `make` does) then it would be fine, but here we are.